master/usage/troubleshooting/index.md

```diff
@@ -3,51 +3,65 @@ title: Troubleshooting
+that the node name for that host is properly configured. The name for the [Node resource](../../reference/calicoctl/resources/node) must match
+the node name in the [workload endpoint resources](../../reference/calicoctl/resources/workloadendpoint) on that host. If the names are mismatched,
+it is likely that all workloads on that node will not receive networking.
 
-```shell
-    sudo ETCD_ENDPOINTS=http://172.25.0.1:2379 calicoctl node run
-```
+To check this, query one of the broken workload endpoints and check its node name:
 
-Also be aware that connection information can be specified as a config
-file rather than using environment variables.  See the
-[calicoctl Configuration Overview]({{site.baseurl}}/{{page.version}}/reference/calicoctl/setup)
-guide for details.
+	calicoctl get workloadendpoints -n <namespace>
 
-## Ubuntu (or GNOME) NetworkManager
+Then, check to see if a single corresponding node resource exists:
 
-Disable [NetworkManager](https://help.ubuntu.com/community/NetworkManager) before
-attempting to use {{site.prodname}} networking.
+	calicoctl get nodes
 
-NetworkManager manipulates the routing table for interfaces in the default network
-namespace where {{site.prodname}} veth pairs are anchored for connections to containers.
-This can interfere with the {{site.prodname}} agent's ability to route correctly.
+If the node resource either does not exist or there are multiple node resources representing the bad node, it is likely that the node's hostname has changed. This often happens
+as a result of switching a node's hostname between its FQDN and its short DNS name.
 
-You can configure interfaces in the `/etc/network/interfaces` file if the
-NetworkManager removes your host's interfaces. See the Debian
-[NetworkConfiguration](https://wiki.debian.org/NetworkConfiguration)
-guide for more information.
+To correct this, you must perform the following steps (with examples shown using Kubernetes):
 
-## etcd.EtcdException: No more machines in the cluster
+1. Prevent new workloads from being scheduled on the bad node.
+```
+	kubectl cordon mynode.internal.projectcalico.org
+```
+1. Drain all workloads from the node.
+```
+	kubectl drain mynode.internal.projectcalico.org --ignore-daemonsets
+```
+1. On the bad node, set the hostname to the desired value.
+``
+	sudo hostnamectl set-hostname <desired-hostname>`
+```
+1. Delete the bad node configuration from Calico
+```
+	calicoctl delete node <name-of-bad-node>
+```
+1. Restart Calico on the bad node to pick up the changes.
+```
+	kubectl delete pod -n kube-system <name-of-calico-pod>
+```
+1. Reenable scheduling of worklods on the node.
+```
+	kubectl uncordon mynode.internal.projectcalico.org
+```
 
-If you see this exception, it means `calicoctl` can't communicate with your etcd
-cluster.  Ensure etcd is up and listening on `localhost:2379`
+To prevent this problem from ocurring, it is recommended to always explicitly specifiy the desired node name
```

Comment:
"ocurring" --> "occurring"
Also, "it is recommended to always explicitly specify" --> "we recommend always explicitly specifying"